lib/rules/max-len.js

```diff
@@ -157,11 +157,14 @@ module.exports = {
          */
         function isFullLineComment(line, lineNumber, comment) {
             var start = comment.loc.start,
-                end = comment.loc.end;
+                end = comment.loc.end,
+                isFirstTokenOnLine = !line.slice(0, comment.loc.start.column).trim(),
```

Comment:
Other rules (such as [lines-around-comment](https://github.com/eslint/eslint/blob/master/lib/rules/lines-around-comment.js#L148) are able to use `sourceCode.getTokenBefore` and could check the location that way - in this rule I only had access to the line, so I thought this was a pretty elegant solution